File: Data Structures & Algorithms/sum-of-all-subset-xor-totals/submission-0.java

class Solution {
    public int subsetXORSum(int[] nums) {
        int n = nums.length;
        int ans = 0;

        // Iterate over all subsets
        for (int mask = 0; mask < (1 << n); mask++) {
            int xor = 0;

            // Check which elements are included
            for (int i = 0; i < n; i++) {
                if ((mask & (1 << i)) != 0) {
                    xor ^= nums[i];
                }
            }

            ans += xor;
        }

        return ans;
    }
}

File: Data Structures & Algorithms/move-zeroes/submission-0.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        int cnt = 0;
        for(int i = 0; i < n; i++) {
            if(nums[i] == 0) {
                cnt++;
            }
        }
        int k = n - cnt;
        vector<int> a(n);
        for(int i = 0;i < n;i++) {
            a[i] = 0;
        }
        int j = 0;
        int i = 0;
        while(k > 0) {
            if(nums[i] != 0) {
                a[j] = nums[i];
                j++;
                k--;
            }
            i++;
            
        }
        for(int i = 0;i < n;i++) {
            nums[i] = a[i];
        }
    }
};